sdks/python/apache_beam/testing/test_stream.py

```diff
@@ -133,15 +140,17 @@ def __eq__(self, other):
     return self.new_watermark == other.new_watermark and self.tag == other.tag
 
   def __hash__(self):
-    return hash(self.new_watermark)
+    return hash(str(self.new_watermark) + str(self.tag))
 
   def __lt__(self, other):
     return self.new_watermark < other.new_watermark
 
   def to_runner_api(self, unused_element_coder):
+    tag = 'None' if self.tag is None else self.tag
     return beam_runner_api_pb2.TestStreamPayload.Event(
         watermark_event=beam_runner_api_pb2.TestStreamPayload.Event.
-        AdvanceWatermark(new_watermark=self.new_watermark.micros // 1000))
+        AdvanceWatermark(
+            new_watermark=self.new_watermark.micros // 1000, tag=tag))
```

Comment:
assert that we aren't losing precision here.

Reply:
done

Reply:
This should be done for ProcessingTimeEvent as well